browser/brave_news/brave_news_tab_helper.cc

```diff
@@ -89,48 +102,109 @@ bool BraveNewsTabHelper::IsSubscribed(const FeedDetails& feed_details) {
 }
 
 bool BraveNewsTabHelper::IsSubscribed() {
-  for (const auto& feed : GetAvailableFeeds()) {
-    if (IsSubscribed(feed))
+  for (const auto& feed_url : GetAvailableFeedUrls()) {
+    if (IsSubscribed(feed_url)) {
       return true;
+    }
   }
   return false;
 }
 
-void BraveNewsTabHelper::ToggleSubscription(const FeedDetails& feed_details) {
-  bool subscribed = IsSubscribed(feed_details);
-  auto* publisher = controller_->publisher_controller()->GetPublisherForFeed(
-      feed_details.feed_url);
+std::string BraveNewsTabHelper::GetTitleForFeedUrl(const GURL& feed_url) {
+  // If there's a default publisher for this |feed_url| we should use it's
+  // title.
+  auto* default_publisher =
+      controller_->publisher_controller()->GetPublisherForFeed(feed_url);
+  if (default_publisher) {
+    return default_publisher->publisher_name;
+  }
+
+  // Otherwise, find the |FeedDetails| for this |feed_url|.
+  auto it = base::ranges::find_if(
+      rss_page_feeds_,
+      [feed_url](const auto& details) { return details.feed_url == feed_url; });
```

Comment:
totally optional: using projection can help us to simply this.

```suggestion
  auto it = base::ranges::find(rss_page_feeds_, feed_url, &FeedDetails::feed_url);
```

Reply:
That's super cool - I really struggle with what I can do with which functions in C++ :laughing: